### PR TITLE
Added empty content message for empty read-only folders

### PR DESCRIPTION
**Pull request by PVince81**
Please review @rullzer @nickvergessen @MorrisJobke @owncloud/designers

### CODE REVIEW

**Comment by PVince81** [comment]
This works for both read-only public shares and read-only regular shares.

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/15508

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -1623,7 +1623,8 @@
 		updateEmptyContent: function() {
 			var permissions = this.getDirectoryPermissions();
 			var isCreatable = (permissions & OC.PERMISSION_CREATE) !== 0;
-			this.$el.find('#emptycontent').toggleClass('hidden', !isCreatable || !this.isEmpty);
+			this.$el.find('#emptycontent').toggleClass('hidden', !this.isEmpty);
+			this.$el.find('#emptycontent .uploadmessage').toggleClass('hidden', !isCreatable || !this.isEmpty);
```
Comment (nickvergessen) [inline apps/files/js/filelist.js]: `!isCreatable ` should be enough here, because `#emptycontent` is not visibile otherwise anyway?

Comment (PVince81) [inline apps/files/js/filelist.js]: True. But I'll leave it for now.

File: apps/files/templates/list.php [hunk 2]
```diff
@@ -54,7 +54,7 @@
 <div id="emptycontent" class="hidden">
 	<div class="icon-folder"></div>
 	<h2><?php p($l->t('No files in here')); ?></h2>
-	<p><?php p($l->t('Upload some content or sync with your devices!')); ?></p>
+	<p class="uploadmessage"><?php p($l->t('Upload some content or sync with your devices!')); ?></p>
```
Comment (nickvergessen) [inline apps/files/templates/list.php]: add class hidden here aswell, so the message doesn't flicker for a short moment, when the permissions are not granted?

Comment (PVince81) [inline apps/files/templates/list.php]: the message will not flicker. The rendering engine only updates what you see once it reaches the JS event loop.
This means that you can make any kind of changes in the UI but the user will only see the result once the JS code reaches the end. (if you want it to flicker, you'd need to use _.defer() or setTimeout())

Comment (PVince81) [inline apps/files/templates/list.php]: Done

**Comment by MorrisJobke** [comment]
Verified broken behaviour and now the fixed state with this PR :+1:

**Comment by jancborchardt** [comment]
Great stuff! :+1: thank you @PVince81 :)

**Comment by karlitschek** [comment]
:+1:

**Comment by PVince81** [comment]
Damn, pushed too early. I'll take this opportunity to squash.

**Comment by PVince81** [comment]
Jenkins PR: https://github.com/owncloud/core/pull/15776

**Pull request merged**